armasec/pytest_extension.py

```diff
@@ -194,26 +204,47 @@ def rs256_openid_config(rs256_iss, rs256_jwks_uri):
     )
 
 
+def build_mock_openid_server(
+    domain, openid_config, jwk, jwks_uri
+) -> Callable[[str, OpenidConfig, JWK, str], _GeneratorContextManager]:
+    """
+    Build a helper function to mock the routes used to load the openid-configuration.
+    """
+
+    @contextmanager
+    def _helper(
+        domain: str = domain,
+        openid_config: OpenidConfig = openid_config,
+        jwk: JWK = jwk,
+        jwks_uri: str = jwks_uri,
+    ):
+        MockOpenidRoutes = namedtuple("MockOpenidRoutes", ["openid_config_route", "jwks_route"])
+        with respx.mock:
+            openid_config_route = respx.get(
+                OpenidConfigLoader.build_openid_config_url(domain),
+            )
+            openid_config_route.return_value = httpx.Response(
+                starlette.status.HTTP_200_OK,
+                json=openid_config.dict(),
+            )
+
+            jwks = JWKs(keys=[jwk])
+            jwks_route = respx.get(jwks_uri)
+            jwks_route.return_value = httpx.Response(
+                starlette.status.HTTP_200_OK,
+                json=jwks.dict(),
+            )
+            yield MockOpenidRoutes(openid_config_route, jwks_route)
+
+    return _helper
+
+
 @pytest.fixture
 def mock_openid_server(rs256_domain, rs256_openid_config, rs256_jwk, rs256_jwks_uri):
     """
     Mock routes that would be used in loading an openid-configuration and jwks from a typical
     openid server.
     """
-    MockOpenidRoutes = namedtuple("MockOpenidRoutes", ["openid_config_route", "jwks_route"])
-    with respx.mock:
-        openid_config_route = respx.get(
-            OpenidConfigLoader.build_openid_config_url(rs256_domain),
-        )
-        openid_config_route.return_value = httpx.Response(
-            starlette.status.HTTP_200_OK,
-            json=rs256_openid_config.dict(),
-        )
-
-        jwks = JWKs(keys=[rs256_jwk])
-        jwks_route = respx.get(rs256_jwks_uri)
-        jwks_route.return_value = httpx.Response(
-            starlette.status.HTTP_200_OK,
-            json=jwks.dict(),
-        )
-        yield MockOpenidRoutes(openid_config_route, jwks_route)
+    builder = build_mock_openid_server(rs256_domain, rs256_openid_config, rs256_jwk, rs256_jwks_uri)
```

Comment:
I think we should use this as a fixture in the function signature of this fixture. Otherwise it looks kinda funny to call it twice:
```python
@pytest.fixture
def mock_openid_server(build_mock_openid_server, ...):
    with build_mock_openid_server(...) as constructed builder:
        yield constructed_builder
```

Reply:
I tried with all my forces but it doesn't work this way. Perhaps we can pair sometime and try to implement this way